### PR TITLE
Assignment7

### PR DESCRIPTION
**Pull request by worseTyler**

### CODE REVIEW

**Comment by ZStuefen** [comment]
I will review this

**Comment by ZStuefen** [comment]
# Assignment
- Using PlaywrightSharp, automate the following items of the website
  - Verify that when landing on the homepage, Secret Santa is displayed in the navigation banner ✔
  - Verify that all 3 pages (Users, Groups, Gifts) can be reached via clicking on the navigation elements ✔
  - Create a Gift (should assert that the number of gifts after the creation is one greater than the number of gifts before the user was created) ✔
  - Modify the last gift in the list (should assert that the updated data was successfully saved/displayed to the screen) ✔
  - Delete the last gift in the list (should assert that the number of gifts after the delete is one less than the number of gifts before the deletion) ✔
  - All of these tests should be able to be executed without having to manually start the Web and Api projects ✔
- Using AlpineJS, convert the CRUD pages for Users to be javascript based instead of MVC based
  - Index page should fetch the Users via javascript directly from the API ✔
  - Create page should submit the Users information via javascript directly to the API. Upon successful save, the user should be redirected back to the Users index page ✔
  - Edit page should submit the Users information via javascript directly to the API. Upon successful update, the user should be redirected back to the Users index page ✔
  - Delete opration should submit the request directly to the API via javascript and then refresh the Index page via a javascript call directly to the API ✔

File: .github/workflows/AssignmentBuild.yml [hunk 1]
```diff
@@ -28,6 +28,9 @@ jobs:
         npm install
         npm run build:prod
 
+    - name: Trust Certificates
```
Comment (breaman) [inline .github/workflows/AssignmentBuild.yml:31]: Nice, going to have to make sure I do this myself.

File: SecretSanta/src/SecretSanta.Web/Views/Users/Index.cshtml [hunk 2]
```diff
@@ -1,18 +1,13 @@
-@model List<UserViewModel>
-
-
-<section class="flex flex-wrap mb-6">
-    @foreach(var userViewModel in Model) {
+<section class="flex flex-wrap mb-6" x-data="SecretSanta.setupUsers()" x-init="mounted()">
+    <template x-for="(currentUser, index) in users" :key="index">
         <section class="w-full sm:w-1/3">
-            <a asp-action="Edit" asp-route-id=@userViewModel.Id>
+            <a :href="'https://localhost:5001/users/edit/' + currentUser.id">
```
Comment (breaman) [inline SecretSanta/src/SecretSanta.Web/Views/Users/Index.cshtml:4]: Not sure why you hardcoded the url to be port 5001 here.

File: SecretSanta/test/SecretSanta.Web.EndToEnd/EndToEndTests.cs [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        [TestMethod]
+        public async Task NavigateBetweenPages()
+        {
+            var page = await Browser.NewPageAsync();
+            var response = await page.GoToAsync(LocalHost);
+
+            Assert.IsTrue(response.Ok);
+
+            await page.ClickAsync("text=Users");
+            Assert.AreEqual<string>(LocalHost + "Users", page.Url);
```
Comment (breaman) [inline SecretSanta/test/SecretSanta.Web.EndToEnd/EndToEndTests.cs:55]: I like this way of validating that each page loaded, nice job.

**Review by breaman — COMMENTED** [review]
Nice work overall on the assignment.

**Pull request closed without merging**